recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+import pathlib
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+from conan.tools.apple import is_apple_os
+from conan.tools.files import apply_conandata_patches, get, copy, rm
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+
+
+required_conan_version = ">=1.53.0"
+
+
+class PackageConan(ConanFile):
+    name = "openassetio"
+    description = "An open-source interoperability standard for tools and content management systems used in media production."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/OpenAssetIO/OpenAssetIO"
+    topics = ("asset-pipeline", "vfx", "cg", "assetmanager", "vfx-pipeline")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "with_python": [True, False],
+        "python_version": ["3.7.12", "3.8.12", "3.9.7", "3.10.0"]
+    }
+    default_options = {
+        "shared": False,
+        "with_python": True,
+        "python_version": "3.9.7"
+    }
+    short_paths = True
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "9",
+            "clang": "12",
+            "apple-clang": "12",
+        }
+
+    def configure(self):
+        if self.options.with_python:
+            if is_msvc(self):
+                # Required to create import .lib for building extension module.
+                self.options["cpython"].shared = True
+        else:
+            self.options.rm_safe("python_version")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("tomlplusplus/3.2.0")
+        if self.options.with_python:
+            # Avoid conflict with cmake vs cpython openssl dependency.
+            self.requires("openssl/1.1.1s")
+            # TODO: cpython requires ncurses/6.2 but no pre-built package exists.
+            self.requires(f"ncurses/6.3")
+            self.requires(f"cpython/{self.options.python_version}")
```

Comment:
I only have one question, I am not sure this should be an option... the version of dependencies is include in the package ID so the option does not add any value.

Users are able to set requirements as well so it's not required for handling conflicts

Reply:
Technically `cpython` is only a build (and test) dependency, for building the Python extension module, which can then work in any compatible Python environment (where 'compatible' means same minor version of Python, usually). 

What would be the alternative to a `python_version` option?  Perhaps pin to a particular `cpython` version, then if a consumer wants a different version they have to explicitly specify `cpython/X.Y.Z` in their `conanfile`/command-line?

Reply:
To expedite things, I've updated to remove the `python_version` option in 118911b0.  Can always drop that commit if we decide against.